Baidu Scholar.js

```diff
@@ -124,16 +125,24 @@ function scrape(doc, _url) {
 			if (!item.creators || item.creators.length == 0) {
 				item.creators = [];
 				doc.querySelectorAll('p.author_text a').forEach((e) => {
-					item.creators.push(ZU.cleanAuthor(e.textContent, 'author', true));
+					var creator = ZU.cleanAuthor(e.textContent, 'author', true);
+					item.creators.push(creator);
+					var lastSpace = creator.lastName.lastIndexOf(' ');
+					if (creator.lastName.search(/[A-Za-z]/) == -1 && lastSpace == -1) {
+						// Chinese name. first character is last name, the rest are first name
```

Comment:
That might be usually the case but not always e.g. 司馬 is a compound last name. See also https://github.com/zotero/translators/pull/2089#issuecomment-568266419

Reply:
@zuphilip Hi, I think it's hard to detect some compound Chinese name. In Chinese,  司馬 and 司 are both last name, another examples like 夏侯 and 夏， 令狐 and 令。The first character of compound Chinese name is common single Chinese last name.  In these cases, I think get the first character as last name can cover the most. But the rest compound Chinese name, I think we can build a array to map.

81 common compound Chinese name (in simplified Chinese), 
欧阳、太史、端木、上官、司马、东方、独孤、南宫、万俟、闻人、夏侯、诸葛、尉迟、公羊、赫连、澹台、皇甫、宗政、濮阳、公冶、太叔、申屠、公孙、慕容、仲孙、钟离、长孙、宇文、司徒、鲜于、司空、闾丘、子车、亓官、司寇、巫马、公西、颛孙、壤驷、公良、漆雕、乐正、宰父、谷梁、拓跋、夹谷、轩辕、令狐、段干、百里、呼延、东郭、南门、羊舌、微生、公户、公玉、公仪、梁丘、公仲、公上、公门、公山、公坚、左丘、公伯、西门、公祖、第五、公乘、贯丘、公皙、南荣、东里、东宫、仲长、子书、子桑、即墨、达奚、褚师

Reply:
Okay, I see. When we see 司馬... is it then also possible that the first name starts with 馬 ? If not, then it could be a strategy to first search for compound last names and only afterwards for simple last names.

> But the rest compound Chinese name, I think we can build a array to map.

What do you mean here with the "rest" exactly? The 81-3 common compound Chinese names or something different?

Reply:
馬 is also a single first name. I think the 81 compound Chinese names can divide into two groups. The first group contains 司马, 诸葛, 夏侯, 令狐, etc. In this group, the first character of the compound name also can be a common Chinese last name. In the second group,  the first character is not a first name or very very few people have the first character as their last name.   

But in my opinion, we can simply use the first character as the first name. The following reasons:
1. You can not make all people satisfied
No matter use the first character or the first two characters as the first name, you always will get a wrong first name.
2. The Chinese first name distribution, the compound first name is the minority
According to an [article](https://kns.cnki.net/KCMS/detail/detail.aspx?dbcode=CJFQ&dbname=CJFD2014&filename=ZGTJ201406011&v=MTIyMjFyQ1VSTE9lWmVScEZpRG1VN3pMUHlyZlpMRzRIOVhNcVk5RVpZUjhlWDFMdXhZUzdEaDFUM3FUcldNMUY=) in CNKI. In the 2010 China National Census, there are 11 compound Chinese first names whose population is larger than 10000. I get two figures from this article.
![image](https://user-images.githubusercontent.com/2743744/71452823-30db3600-27c3-11ea-86a0-8dc5872f5152.png)

Figure 1 shows the top 100 Chinese first name. These first names in the table are all single first name. The amount of the top 100 Chinese first names covers 82.1% of the total population of China.

![](https://user-images.githubusercontent.com/2743744/71453507-79e0b980-27c6-11ea-8891-f5e7c63c3aed.png)

Figure 2 shows the top 11 compound first name(population larger than ten thousand).  The most common compound first name is 欧阳 with a population of  901000.

Use the first character as Chinese first name, the code maintenance is easy. When an item exists compound first name,  they can edit it manually in Zotero.  

I hope this information will help you.

Reply:
Thank you for all these information. That is interesting to know also besides Zotero context. I tried to make a [query in Wikidata](https://query.wikidata.org/#%23%20For%20which%20Chinese%20compound%20last%20names%20is%20the%20first%20character%20only%20a%20last%20name%20as%20well%3F%0ASELECT%20%2a%20WHERE%20%7B%0A%20%20VALUES%20%3FcompoundName%20%7B%0A%20%20%20%20%20%22%E6%AC%A7%E9%98%B3%22%40zh%20%22%E5%A4%AA%E5%8F%B2%22%40zh%20%22%E7%AB%AF%E6%9C%A8%22%40zh%20%22%E4%B8%8A%E5%AE%98%22%40zh%20%22%E5%8F%B8%E9%A9%AC%22%40zh%20%22%E4%B8%9C%E6%96%B9%22%40zh%20%22%E7%8B%AC%E5%AD%A4%22%40zh%20%22%E5%8D%97%E5%AE%AB%22%40zh%20%22%E4%B8%87%E4%BF%9F%22%40zh%20%22%E9%97%BB%E4%BA%BA%22%40zh%20%0A%20%20%20%20%20%22%E5%A4%8F%E4%BE%AF%22%40zh%20%22%E8%AF%B8%E8%91%9B%22%40zh%20%22%E5%B0%89%E8%BF%9F%22%40zh%20%22%E5%85%AC%E7%BE%8A%22%40zh%20%22%E8%B5%AB%E8%BF%9E%22%40zh%20%22%E6%BE%B9%E5%8F%B0%22%40zh%20%22%E7%9A%87%E7%94%AB%22%40zh%20%22%E5%AE%97%E6%94%BF%22%40zh%20%22%E6%BF%AE%E9%98%B3%22%40zh%20%22%E5%85%AC%E5%86%B6%22%40zh%0A%20%20%20%20%20%22%E5%A4%AA%E5%8F%94%22%40zh%20%22%E7%94%B3%E5%B1%A0%22%40zh%20%22%E5%85%AC%E5%AD%99%22%40zh%20%22%E6%85%95%E5%AE%B9%22%40zh%20%22%E4%BB%B2%E5%AD%99%22%40zh%20%22%E9%92%9F%E7%A6%BB%22%40zh%20%22%E9%95%BF%E5%AD%99%22%40zh%20%22%E5%AE%87%E6%96%87%22%40zh%20%22%E5%8F%B8%E5%BE%92%22%40zh%20%22%E9%B2%9C%E4%BA%8E%22%40zh%0A%20%20%20%20%20%22%E5%8F%B8%E7%A9%BA%22%40zh%20%22%E9%97%BE%E4%B8%98%22%40zh%20%22%E5%AD%90%E8%BD%A6%22%40zh%20%22%E4%BA%93%E5%AE%98%22%40zh%20%22%E5%8F%B8%E5%AF%87%22%40zh%20%22%E5%B7%AB%E9%A9%AC%22%40zh%20%22%E5%85%AC%E8%A5%BF%22%40zh%20%22%E9%A2%9B%E5%AD%99%22%40zh%20%22%E5%A3%A4%E9%A9%B7%22%40zh%20%22%E5%85%AC%E8%89%AF%22%40zh%0A%20%20%20%20%20%22%E6%BC%86%E9%9B%95%22%40zh%20%22%E4%B9%90%E6%AD%A3%22%40zh%20%22%E5%AE%B0%E7%88%B6%22%40zh%20%22%E8%B0%B7%E6%A2%81%22%40zh%20%22%E6%8B%93%E8%B7%8B%22%40zh%20%22%E5%A4%B9%E8%B0%B7%22%40zh%20%22%E8%BD%A9%E8%BE%95%22%40zh%20%22%E4%BB%A4%E7%8B%90%22%40zh%20%22%E6%AE%B5%E5%B9%B2%22%40zh%20%22%E7%99%BE%E9%87%8C%22%40zh%0A%20%20%20%20%20%22%E5%91%BC%E5%BB%B6%22%40zh%20%22%E4%B8%9C%E9%83%AD%22%40zh%20%22%E5%8D%97%E9%97%A8%22%40zh%20%22%E7%BE%8A%E8%88%8C%22%40zh%20%22%E5%BE%AE%E7%94%9F%22%40zh%20%22%E5%85%AC%E6%88%B7%22%40zh%20%22%E5%85%AC%E7%8E%89%22%40zh%20%22%E5%85%AC%E4%BB%AA%22%40zh%20%22%E6%A2%81%E4%B8%98%22%40zh%20%22%E5%85%AC%E4%BB%B2%22%40zh%0A%20%20%20%20%20%22%E5%85%AC%E4%B8%8A%22%40zh%20%22%E5%85%AC%E9%97%A8%22%40zh%20%22%E5%85%AC%E5%B1%B1%22%40zh%20%22%E5%85%AC%E5%9D%9A%22%40zh%20%22%E5%B7%A6%E4%B8%98%22%40zh%20%22%E5%85%AC%E4%BC%AF%22%40zh%20%22%E8%A5%BF%E9%97%A8%22%40zh%20%22%E5%85%AC%E7%A5%96%22%40zh%20%22%E7%AC%AC%E4%BA%94%22%40zh%20%22%E5%85%AC%E4%B9%98%22%40zh%0A%20%20%20%20%20%22%E8%B4%AF%E4%B8%98%22%40zh%20%22%E5%85%AC%E7%9A%99%22%40zh%20%22%E5%8D%97%E8%8D%A3%22%40zh%20%22%E4%B8%9C%E9%87%8C%22%40zh%20%22%E4%B8%9C%E5%AE%AB%22%40zh%20%22%E4%BB%B2%E9%95%BF%22%40zh%20%22%E5%AD%90%E4%B9%A6%22%40zh%20%22%E5%AD%90%E6%A1%91%22%40zh%20%22%E5%8D%B3%E5%A2%A8%22%40zh%20%22%E8%BE%BE%E5%A5%9A%22%40zh%0A%20%20%20%20%20%22%E8%A4%9A%E5%B8%88%22%40zh%0A%20%20%7D%0A%20%20BIND%28SUBSTR%28%3FcompoundName%2C%201%2C%201%29%20AS%20%3FfirstCharacter%29%0A%20%20%3Fitem%20wdt%3AP1705%20%3FfirstCharacter%20.%0A%20%20FILTER%20EXISTS%20%7B%0A%20%20%20%20%3Fitem%20wdt%3AP31%2Fwdt%3AP279%2a%20wd%3AQ101352%20.%0A%20%20%7D%0A%7D) about the group 1 (compound names where the first character can be a family name as well) and found some more examples.

I see now that we cannot do much here with regard to the name splitting for compound Chinese names. However, on the other hand as you outlined, taking the first character for the last name will work for all single Chinese last names and this is a large majority of the people. Even the compound names starts with this character and therefore maybe a good start as well there.

Thank you again for all the information and your patience explaining this to me! :bowing_man:

Reply:
Zotero is great, it's my pleasure to make it better. 

May I delete my fork?